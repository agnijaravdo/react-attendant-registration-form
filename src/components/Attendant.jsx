const Attendant = ({ attendant: { name, lastName, jobTitle, age } }) => (
  <div data-testid="attendant">
    <br />
    Name: <span className="attendants-info">{name} </span>
    Last Name: <span className="attendants-info">{lastName} </span>
    Job Title: <span className="attendants-info">{jobTitle} </span>
    Age: <span className="attendants-info">{age} </span>
    <br />
  </div>
);

export default Attendant;
