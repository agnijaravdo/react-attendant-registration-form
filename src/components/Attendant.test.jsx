import Attendant from "./Attendant";
import { render, screen } from "@testing-library/react";

describe("<Attendant />", () => {
  let props;

  beforeEach(() => {
    props = {
      attendant: {
        name: "John",
        lastName: "Doe",
        jobTitle: "Engineer",
        age: 40,
      },
    };
  });

  it("renders correctly on successful initial load", async () => {
    const { container } = render(<Attendant {...props} />);

    expect(container).toMatchSnapshot();
  });

  it("renders component with given props", async () => {
    render(<Attendant {...props} />);

    expect(await screen.findByText("John")).toBeInTheDocument();
    expect(await screen.findByText("Doe")).toBeInTheDocument();
    expect(await screen.findByText("Engineer")).toBeInTheDocument();
    expect(await screen.findByText("40")).toBeInTheDocument();
  });
});
