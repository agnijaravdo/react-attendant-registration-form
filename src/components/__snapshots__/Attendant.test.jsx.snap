// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Attendant /> renders correctly on successful initial load 1`] = `
<div>
  <div>
    <br />
    Name: 
    <span
      class="attendants-info"
    >
      John
       
    </span>
    Last Name: 
    <span
      class="attendants-info"
    >
      Doe
       
    </span>
    Job Title: 
    <span
      class="attendants-info"
    >
      Engineer
       
    </span>
    Age: 
    <span
      class="attendants-info"
    >
      40
       
    </span>
    <br />
  </div>
</div>
`;
